src/test/librados/io.cc

```diff
@@ -826,6 +827,12 @@ TEST_F(LibRadosIoEC, OverlappingWriteRoundTrip) {
   char *buf = (char *)new char[dbsize];
   char *buf2 = (char *)new char[bsize];
   char *buf3 = (char *)new char[dbsize];
+  auto lam = [&] {
+            delete[] buf;
+            delete[] buf2;
+            delete[] buf3;
+          };
+  scope_guard<decltype(lam)> sg(std::forward<decltype(lam)>(lam));
```

Comment:
Yes, good question. The reason is that cppcheck issues an error about a memory leak (false positive) when make_scope_guard is used but does not when creating a scope_guard object with a named lambda. I have reported this upstream and am trying to get access to create a cppcheck issue.